### PR TITLE
Bump used RPC to version 0.7.0

### LINKED ISSUE #1857: Bump used RPC to 0.7.0
### Which components does the task require to be changed? (think hard pls)

snforge

### Description

We need to bump `starknet-rs` to the one that supports RPC v0.7.0
The 0.7.0 contains breaking changes so it needs to be updated

Ideally this should be released with 0.20.0

### PR DESCRIPTION
**Pull request by Draggu**
Closes #1857

## Introduced changes

- Bumped used RPC version to `0.7.0-rc2`

## Checklist

- [x] Linked relevant issue
- [x] Updated relevant documentation
- [x] Added relevant tests
- [x] Performed self-review of the code
- [x] Added changes to `CHANGELOG.md`

### CODE REVIEW

**Comment by Draggu** [comment]
Blocked by https://github.com/foundry-rs/starknet-foundry/pull/1851. Changes made in https://github.com/foundry-rs/starknet-foundry/pull/1851 are required for this to pass.

File: scripts/install_devnet.sh
Comment (ddoktorski) [inline scripts/install_devnet.sh:10]: I think these variables are not needed anymore.

File: crates/sncast/tests/helpers/devnet.rs
Comment (ddoktorski) [inline crates/sncast/tests/helpers/devnet.rs:43]: Note: This is necessary, because devnet sets `cairo1` as the default from the last version. It can be removed once we support Cairo >=1 accounts in #1851

File: crates/forge/tests/data/forking/.gitignore
Comment (ddoktorski) [inline crates/forge/tests/data/forking/.gitignore:2]: nit: Newline

File: crates/cheatnet/tests/cheatcodes/prank.rs
Comment (drknzz) [inline crates/cheatnet/tests/cheatcodes/prank.rs:303]: Why don't we have a constant for this yet 👁️👄👁️

File: crates/sncast/tests/e2e/script/call.rs
Comment (drknzz) [inline crates/sncast/tests/e2e/script/call.rs:105]: Is this right? The previous error seemed different than this one, so maybe just double check

File: crates/sncast/tests/e2e/script/deploy.rs
Comment (drknzz) [inline crates/sncast/tests/e2e/script/deploy.rs:185]: Probably don't want to assert hashes here too, so it won't bother in next update

File: crates/sncast/tests/e2e/script/invoke.rs
Comment (drknzz) [inline crates/sncast/tests/e2e/script/invoke.rs:164]: Same here

File: crates/sncast/tests/e2e/script/call.rs
Comment (THenry14) [reply crates/sncast/tests/e2e/script/call.rs:105]: I think this should be ok, this seems to be indicating invalid calldata

File: crates/sncast/tests/e2e/script/deploy.rs
Comment (THenry14) [reply crates/sncast/tests/e2e/script/deploy.rs:185]: Definitely, lets try to `[..]` them

File: crates/shared/src/consts.rs
Comment (THenry14) [inline crates/shared/src/consts.rs:1]: nitpick: I think a final version of 0.7.0 was released today, it does not seem to contain any changes to rc, but still we should probably consider updating when possible

File: crates/sncast/tests/e2e/script/deploy.rs
Comment (THenry14) [inline crates/sncast/tests/e2e/script/deploy.rs:63]: nit: is it possible to get rid of this? I thought `assert_stdout_contains` can be used to check if output contains some sequence, so we get to choose what we want to match. scarb output is something we do not need to assert in our tests

Comment (THenry14) [reply crates/sncast/tests/e2e/script/deploy.rs:63]: yeah, maybe. 
how come it did not complain earlier, do you happen to know?

File: crates/cheatnet/tests/cheatcodes/prank.rs
Comment (piotmag769) [reply crates/cheatnet/tests/cheatcodes/prank.rs:303]: We have multiple ones even xD We can prob clean it up but not in this PR

File: crates/sncast/tests/e2e/script/deploy.rs
Comment (piotmag769) [reply crates/sncast/tests/e2e/script/deploy.rs:63]: Good question actually since the output inheriting from Scarb didn't change 🤔

File: crates/shared/src/consts.rs
Comment (Draggu) [reply crates/shared/src/consts.rs:1]: https://github.com/foundry-rs/starknet-foundry/issues/1890

**Review by THenry14 — APPROVED** [review]
lgtm, let's wait for an approval from someone from forge team before merge

**Review by piotmag769 — APPROVED** [review]
LGTM, when will our nodes be updated to 0.7.0 so we can bump it ?

**Pull request merged**
